third_party/move/move-vm/types/src/value_serde.rs

```diff
@@ -23,15 +20,6 @@ use std::cell::RefCell;
 /// An extension to (de)serialize information about function values.
 #[cfg_attr(test, automock)]
 pub trait FunctionValueExtension {
-    /// Given the module's id and the function name, returns the parameter types of the
-    /// corresponding function, instantiated with the provided set of type tags.
-    fn get_function_arg_tys(
-        &self,
-        module_id: &ModuleId,
-        function_name: &IdentStr,
-        ty_arg_tags: Vec<TypeTag>,
-    ) -> PartialVMResult<Vec<Type>>;
-
     /// Create an implementation of an `AbstractFunction` from the serialization data.
     fn create_from_serialization_data(
```

Comment:
@gelash these 2 methods are used for function values now, the one above is not needed because we store layouts

Reply:
I'm not quite understanding this. The layout doesn't allow to compute `Type`. The `TypeTag` is stored with the closure, and the stored layout is instantiated. How can you get rid of this?

Reply:
I am saying that `get_function_arg_tys` is not needed (it is also not used) because you added layouts for captured arguments which are checked in paranoid mode when the closure is resolved.